tools/checkswitch/checkswitch.go

```diff
@@ -146,8 +172,66 @@ func run(pass *analysis.Pass) (any, error) {
 						}
 					}
 				}
-			}
 
+			case *ast.SwitchStmt:
+				var name string
+				for _, el := range n.Body.List {
+					if len(el.(*ast.CaseClause).List) < 1 {
+						continue
+					}
+					firstTypeCase := el.(*ast.CaseClause).List[0]
+					switch firstTypeCase := firstTypeCase.(type) {
+					case *ast.StarExpr:
+						if sexp, ok := firstTypeCase.X.(*ast.SelectorExpr); ok {
+							name = sexp.Sel.Name
+						}
+						if sexp, ok := firstTypeCase.X.(*ast.Ident); ok {
+							name = sexp.Name
+						}
+					case *ast.SelectorExpr:
+						name = firstTypeCase.Sel.Name
+
+					case *ast.Ident:
+						name = firstTypeCase.Name
+					}
+					name = strings.ToLower(name)
```

Comment:
Let's not convert it to the lower case, `orderTags` can contain the exact match such as
```
"TagFloat64": 1
```
instead of
```
"tagfloat64": 1
```

Reply:
Got it, but the problem is we are using `tagFloat64` in [decode.go](https://github.com/FerretDB/FerretDB/blob/v1.21.0/internal/bson/decode.go#L63) as mentioned
```
tagFloat64 (not exported)
```
If I update the map such as below, the key won't match as the `T` in TagFloat64 is capital. 
```
"TagFloat64": 1
```

So, can I export all the tag constants in the [tag.go ](https://github.com/FerretDB/FerretDB/blob/v1.21.0/internal/bson/tag.go#L22-L44)..?

Reply:
I think I'm following, so unexported `tagFloat64` is the actual `bson` implementation, and exported `TagFloat64` is used only for test. Is there a reason why we need an exported stud such as `TagFloat64` just for test?  :thinking: 

Could we test unexported tags instead by having `tags.go` in same as this package and unexport them, then `orderTags` could have below?

```
"tagFloat64": 1
```